lib/core/src/lib/datatable/components/boolean-cell/boolean-cell.component.spec.ts

```diff
@@ -0,0 +1,179 @@
+            });
+
+            it('exact string is provided', () => {
+                component.value$.next('true');
+                fixture.detectChanges();
+
+                const booleanCell = getBooleanCell();
+
+                expect(booleanCell).toBeTruthy();
+                expect(booleanCell.textContent.trim()).toBe('true');
+            });
+        });
+
+        describe('should render "false" inside cell when', () => {
+            it('boolean value is false', () => {
+                component.value$.next(false);
+                fixture.detectChanges();
+
+                const booleanCell = getBooleanCell();
+
+                expect(booleanCell).toBeTruthy();
+                expect(booleanCell.textContent.trim()).toBe('false');
+            });
+
+            it('exact string is provided', () => {
+                component.value$.next('false');
+                fixture.detectChanges();
+
+                const booleanCell = getBooleanCell();
+
+                expect(booleanCell).toBeTruthy();
+                expect(booleanCell.textContent.trim()).toBe('false');
+            });
+        });
+
+        describe('should NOT render value inside cell in case of', () => {
+            it('invalid string', () => {
+                component.value$.next('tru');
+                fixture.detectChanges();
+
+                const booleanCell = getBooleanCell();
+
+                expect(booleanCell).toBeFalsy();
+            });
+
+            it('number', () => {
+                component.value$.next(0);
+                fixture.detectChanges();
+
+                const booleanCell = getBooleanCell();
+
+                expect(booleanCell).toBeFalsy();
+            });
+
+            it('object', () => {
+                component.value$.next({});
+                fixture.detectChanges();
+
+                const booleanCell = getBooleanCell();
+
+                expect(booleanCell).toBeFalsy();
+            });
+
+            it('null', () => {
+                component.value$.next(null);
+                fixture.detectChanges();
+
+                const booleanCell = getBooleanCell();
+
+                expect(booleanCell).toBeFalsy();
+            });
+
+            it('undefined', () => {
+                component.value$.next(undefined);
+                fixture.detectChanges();
+
+                const booleanCell = getBooleanCell();
+
+                expect(booleanCell).toBeFalsy();
+            });
+
+            it('empty string', () => {
+                component.value$.next('');
+                fixture.detectChanges();
+
+                const booleanCell = getBooleanCell();
+
+                expect(booleanCell).toBeFalsy();
+            });
+
+            it('NaN', () => {
+                component.value$.next(NaN);
+                fixture.detectChanges();
+
+                const booleanCell = getBooleanCell();
+
+                expect(booleanCell).toBeFalsy();
```

Comment:
you can extract this logic to a function that will check the expected result, to avoid code duplication (and make sonar lint happy :P )
in the function, you can pass value (e.g. NaN), and result (false);

```suggestion
   const renderAndCheckValue = (value: any, expectedResult: boolean) => {
        component.value$.next(value);
        
        fixture.detectChanges();
        const booleanCell = getBooleanCell();

        expect(booleanCell).toBe(expectedResult);
    };
```